frontend/src/utils/getCurrentDeviceId/getCurrentDeviceId.tsx

```diff
@@ -0,0 +1,41 @@
+let cachedDevices: MediaDeviceInfo[] = [];
```

Comment:
Keeping stuff in file storage like this will make it harder to test. Prefer to create stuff in closure and create it via a factory function. Then in the react interface you can use a ref as a holder to ensure its only created once